azuredeploy.json

```diff
@@ -265,8 +279,7 @@
     "feSubnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets/', variables('networkSettings').virtualNetworkName, variables('networkSettings').feSubnetName)]",
     "felbPublicIPAddressID": "[resourceId('Microsoft.Network/publicIPAddresses',variables('networkSettings').felbPublicIPAddressName)]",
     "chefAutoPublicIPAddressID": "[resourceId('Microsoft.Network/publicIPAddresses',variables('networkSettings').chefAutoPublicIPAddressName)]",
-    "prefix": "[uniqueString(resourceGroup().id)]",
-    "provider": "[toUpper('33194f91-eb5f-4110-827a-e95f640a9e46')]"
+    "prefix": "[uniqueString(resourceGroup().id)]"
```

Comment:
Using just the `resourceGroup().id` to generate a unique string will not be enough. This is because if someone creates the same Resource Group name then their unique string will be the same as yours.

This is less of a problem now that Storage Accounts are not required as much as they used to be, however it is still good to increase the enrtopy a bit.

I have used this to create the unique string, which is then based on the subscription id, deployment name and the resource group. It can be further increased by adding a prefix, for example:

```
uniqueString(subscription().subscriptionId, resourceGroup().id, deployment().name, parameters('prefix'))
```
```

Reply:
Hi Russell.  I've fixed this and committed.